pkg/executor/scraper/filesystem_extractor.go

```diff
@@ -1,29 +1,128 @@
 package scraper
 
 import (
+	"bytes"
 	"context"
+	"io"
 	"os"
 	"path/filepath"
 
+	"github.com/kubeshop/testkube/pkg/archive"
 	"github.com/kubeshop/testkube/pkg/log"
 
 	"github.com/kubeshop/testkube/pkg/filesystem"
 
 	"github.com/pkg/errors"
 )
 
-type FilesystemExtractor struct {
-	dirs []string
-	fs   filesystem.FileSystem
+type ArchiveFilesystemExtractor struct {
+	fs filesystem.FileSystem
 }
 
-func NewFilesystemExtractor(dirs []string, fs filesystem.FileSystem) *FilesystemExtractor {
-	return &FilesystemExtractor{dirs: dirs, fs: fs}
+func NewArchiveFilesystemExtractor(fs filesystem.FileSystem) *ArchiveFilesystemExtractor {
+	return &ArchiveFilesystemExtractor{fs: fs}
 }
 
-func (e *FilesystemExtractor) Extract(ctx context.Context, process ProcessFn) error {
-	log.DefaultLogger.Infof("extracting files from directories: %v", e.dirs)
-	for _, dir := range e.dirs {
+func (e *ArchiveFilesystemExtractor) Extract(ctx context.Context, paths []string, process ProcessFn) error {
+	log.DefaultLogger.Infof("extracting files from directories: %v", paths)
```

Comment:
Do files need to be "extracted" from directory? It suggest that "extractor" is to extract from archives.